spec/factories/samples.rb

```diff
@@ -34,6 +34,16 @@
     end
 
     after :create do |sample, options|
+      if MetadataField.table_exists?
+        # So it turns out that common metadata fields like collection_location
+        # and sample_type that are already present in the test db
+        # have 'default_for_new_host_genome' set to '0' for some reason.
+        # so trying to update the host genome with
+        # 'sample.host_genome.metadata_fields = MetadataField.where(default_for_new_host_genome: 1)'
+        # only works if you make up a new field. So just shove them all in for now.
+        sample.host_genome.metadata_fields = MetadataField.all
```

Comment:
I'm a little confused by these comments. Is the point just to ensure that there's some MetadataField on the sample?

Reply:
Yeah, not sure what problem you're trying to solve here. Were your tests failing without this line?

Reply:
Yeah, so before when I tried to create a sample with metadata fields in the Rspec tests that use common metadata fields like `collection_location`, the test would not work. When the sample factory tried to create metadatum (the lines after this one) it would fail with the error `Invalid field for host genome Invalid field for metadata key`. I found through some debugging that this was because of some validation code in `metadata.rb`:

```
    # Check if the key is valid. Metadata_field was supposed to be set.
    valid_keys = sample.host_genome.metadata_fields.pluck(:name, :display_name).flatten
    unless key && valid_keys.include?(key) && metadata_field
      errors.add(:invalid_field_for_host_genome, MetadataValidationErrors::INVALID_FIELD_FOR_HOST_GENOME)
      return
    end
```

And when I looked, the `host_genome` being created in the `samples` factory did not have any `metadata_fields`, thus failing the validation. Passing brand new metadata fields in the test would work because they are created with `default_for_new_host_genome: 1` and so are added to the `host_genome` created in the factory. However all the existing metadata fields like `collection_location` and `sample_type` had `default_for_new_host_genome` set to `0`, causing tests using those fields to fail.

Updating them in the rails console worked, but since a new test db created with `docker-compose run --rm web "RAILS_ENV=test rake db:drop db:create db:migrate db:seed"` I was worried that they wouldn't pass the Travis CI tests either.

Admittedly this is not the most elegant solution, especially looking at it now. It might be more correct to look at the transient options and just add the metadata fields passed in. Or even better, change the test db so that common metadata fields don't have `default_for_new_host_genome: 0` by default